### PR TITLE
Fixed missing messages and added evidence

### PR DESCRIPTION
**Pull request by psiinon**
The loops were invalid meaning that no error messages were added. Also added evidence and a simple unit test.

### CODE REVIEW

**Comment by kingthorin** [comment]
DOH missed the build fail when I approved.

**Comment by psiinon** [comment]
Weird, that test works for me locally :/ Will investigate...

**Comment by thc202** [comment]
Travis CI config needs to be changed to use Java 8, because of salvation library:
```
    [junit] Testcase: exampleBadCsp took 0.039 sec
    [junit] 	Caused an ERROR
    [junit] com/shapesecurity/salvation/data/Origin : Unsupported major.minor version 52.0
    [junit] java.lang.UnsupportedClassVersionError: com/shapesecurity/salvation/data/Origin : Unsupported major.minor version 52.0
    [junit] 	at java.lang.ClassLoader.defineClass1(Native Method)
    [junit] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
    [junit] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    [junit] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
    [junit] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
    [junit] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
    [junit] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    [junit] 	at java.security.AccessController.doPrivileged(Native Method)
    [junit] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
    [junit] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
    [junit] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    [junit] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
    [junit] 	at org.zaproxy.zap.extension.cspscanner.ContentSecurityPolicyScannerUnitTest.createScanner(ContentSecurityPolicyScannerUnitTest.java:37)
    [junit] 	at org.zaproxy.zap.extension.pscanrulesAlpha.PassiveScannerTest.setUp(PassiveScannerTest.java:60)
    [junit] 	at org.mockito.internal.runners.JUnit45AndHigherRunnerImpl.run(JUnit45AndHigherRunnerImpl.java:37)
    [junit] 	at org.mockito.runners.MockitoJUnitRunner.run(MockitoJUnitRunner.java:62)
    [junit] 
```

File: src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java [hunk 1]
```diff
@@ -177,17 +183,17 @@ private String getCSPNoticesString(ArrayList<Notice> notices) {
 		ArrayList<Notice> errorsList = Notice.getAllErrors(notices);
 		if (!errorsList.isEmpty()) {
 			returnSb.append(Constant.messages.getString(MESSAGE_PREFIX + "notices.errors")).append(NEWLINE);
-			for (int idx = 0; idx == errorsList.size() - 1; idx++) {
-				returnSb.append(errorsList.get(idx).show()).append(NEWLINE);
+			for (Notice notice : errorsList) {
+				returnSb.append(notice.message).append(NEWLINE);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java]: Was the `show()` dropped on purpose? The exact location of the error could be handy?

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java]: I haven't checked the difference in behavior using `message` versus `show()`. Usage of `show()` came from the examples on their github pages.

Comment (thc202) [inline src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java]: The only difference is that `show()` might include the position/index of the issue in the CSP string.

Comment (psiinon) [inline src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java]: My bad - I thought that was the original bug before I spotted the problems with the loops. show() does give more info so I'll put it back...

**Comment by psiinon** [comment]
Reverted to use show() and changed travis to use java 8, and the tests now pass :)

File: src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java [hunk 2]
```diff
@@ -196,8 +202,8 @@ private String getCSPNoticesString(ArrayList<Notice> notices) {
 		ArrayList<Notice> infoList = Notice.getAllInfos(notices);
 		if (!infoList.isEmpty()) {
 			returnSb.append(Constant.messages.getString(MESSAGE_PREFIX + "notices.infoitems")).append(NEWLINE);
-			for (int idx = 0; idx == infoList.size() - 1; idx++) {
-				returnSb.append(infoList.get(idx).show()).append(NEWLINE);
+			for (Notice notice : warnList) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java]: There was a change in the variable, should be `infoList`.

**Comment by psiinon** [comment]
Good catch - fixed

**Comment by thc202** [comment]
Thanks!

**Pull request merged**